text/0162-refactoring-support.md

```diff
@@ -0,0 +1,606 @@
+     1/3 | 2:03:18 PM | REFACTOR_IN_PROGRESS | AWS::Lambda::Function         | MyStack.FunctionA5EA2BD8     | Service.Function8F0BB69B     
+     2/3 | 2:03:19 PM | REFACTOR_COMPLETE    | AWS::CloudFront::Distribution | MyStack.DistributionE3BB089E | Web.DistributionE3BB089E 
+     3/3 | 2:03:20 PM | REFACTOR_COMPLETE    | AWS::Lambda::Function         | MyStack.FunctionA5EA2BD8     | Service.FunctionA5EA2BD8     
+    
+    ✅  Stack refactor complete
+
+If you pass any filters to the `deploy` command, the refactor will work on those
+stacks plus any other stacks the refactor touches. For example, if you choose to
+only deploy stack A, and a resource was moved from stack A to stack B, the
+refactor will involve both stacks. But even if there was a rename in, let's say,
+stack C, it will not be refactored.
+
+If you want to execute only the automatic refactoring, use the `cdk refactor`
+command. The behavior is basically the same as with `cdk deploy`: it will detect
+whether there are refactors to be made, ask for confirmation if necessary
+(depending on the flag values), and refactor the stacks involved. But it will
+stop there and not proceed with the deployment. If you only want to see what
+changes would be made, use the `--dry-run` flag.
+
+To perform refactoring, the CLI needs new permissions in the bootstrap stack.
+Before using this feature, run `cdk bootstrap` for every target environment, to
+add these new permissions.
+
+Please note that the same CDK application can have multiple stacks for different
+environments. In that case, the CLI will group the stacks by environment and
+perform the refactoring separately in each one. So, although you can move
+resources between stacks, both stacks involved in the move must be in the same
+environment. Trying to move resources across environments will result in an
+error.
+
+### Settings
+
+The behavior of the refactoring feature can be controlled by a few settings.
+
+For both `deploy` and `refactor`:
+
+- `--export-mapping=<FILE>`: writes the mapping to a file. The file can be used
+  later to apply the same refactors to other environments.
+- `--import-mapping=<FILE>`: use the mapping from a file, instead of computing
+  it. The file can be generated using the `--export-mapping` option.
+- `--dry-run`: shows on stdout what would happen, but doesn't execute.
+- `--unstable=refactor`: enables the feature. If the flag is not set, the
+  command fails with an error message explaining that the feature is
+  experimental.
+
+For `deploy` only:
+
+- `--refactoring-action=[ACTION]`: the action to take in case there is a
+  refactor to be made. Possible values for `ACTION` are:
+    - `confirm`: ask the user what to do. This is the scenario described in the
+      **How it works** section.
+    - `refactor`: automatically refactor and deploy.
+    - `quit`: stop with a non-zero exit code.
+    - `skip`: deploy without refactoring. This is the default value.
```

Comment:
Will this always be the default? I would expect most use cases not want this to be the default and either default to refactor or confirm; and choosing this as the default is to not affect existing behavior. Would the plan be to change the default to refactor in the next major version of cdk cli and this functionality is no longer experimental